File: Greedy/1877. Minimize Maximum Pair Sum in Array/1877. Minimize Maximum Pair Sum in Array.py
# Problem          : 1877. Minimize Maximum Pair Sum in Array
# Time Complexity  : O(log n)
# C++ / Python Solution

class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        ans = 0

        for i in range(n // 2):
            ans = max(ans, nums[i] + nums[n - 1 - i])
        return ans

File: Greedy/1877. Minimize Maximum Pair Sum in Array/1877. Minimize Maximum Pair Sum in Array.cpp
# Problem          : 1877. Minimize Maximum Pair Sum in Array
# Time Complexity  : O(log n)
# C++ Solution

class Solution {
public:
    int minPairSum(vector<int>& nums) {
        int n = (int)nums.size();
        sort(nums.begin(), nums.end());
        int ans = 0;
        for (int i = 0; i < n; i++) {
            ans = max(ans, nums[i] + nums[n - 1 - i]);
        }
        return ans;
    }
};